### DISCUSSION TITLE
Could not find plug-in dynamic library - Unity/Wwise

### DISCUSSION THREAD

**Discussion by francescoameglio** [Wasted-Audio/hvcc#203] [Q&A]
Hello, 

I've compiled a PD patch into a wwise plugin that is correctly playing in the wwise project but when I launch the game in Unity I get the following error: 

![immagine](https://github.com/user-attachments/assets/a38040c3-16a5-475d-8c4f-b3fb9a6ef9ae)

Am I missing a step here? Should I do something in Unity as well? 

Thanks

**Comment by francescoameglio** [comment]
I think I've found out what's wrong here: I've forgot to copy the dll in the “DSP” folder to the game project directory (Assets->Wwise->Deployment->Plugin->Windows->x86_64->DSP). It now works.
The only thing is that I've running the plugin as debug config and not release, how do I change this? 
![immagine](https://github.com/user-attachments/assets/97ab9266-460a-4aa8-8e3e-469553676fed)

**Reply by eu-ch** [reply]
Your solution is a bit problematic due to the way Wwise works. Plugins consist of two parts: the Authoring part that manages GUI and some advanced things that Heavy doesn't need, and the Engine part that does all the DSP. The binary you copied from `Authoring/x64/Release/bin/Plugins` contains both parts and is intended for use exclusively by Authoring Tool. You shouldn't use this .dll inside your game.

You need to find a .dll that contains only the Engine part of the plugin. These are placed into `$WWISEROOT/SDK/$Platform` directories. I haven't worked with Unity for quite some time and don't remember where respective .dlls must be copied to.

Also, check out this article: https://www.audiokinetic.com/en/library/edge/?source=Unity&id=pg_install_addlicensedplugins.html
